src/Orleans/Runtime/GrainReference.cs

```diff
@@ -136,9 +136,9 @@ internal static GrainReference NewObserverGrainReference(GrainId grainId, GuidId
         /// <summary>
         /// Called from generated code.
         /// </summary>
-        public static Task<GrainReference> CreateObjectReference(IAddressable o, IGrainMethodInvoker invoker)
+        public static GrainReference CreateObjectReference(IAddressable o, IGrainMethodInvoker invoker)
```

Comment:
I think you can get rid of this method completely and just call `RuntimeClient.Current.CreateObjectReference(o, invoker)` directly where this was called before. It was probably forgotten in some refactoring.